File: elixir/foo/test/foo/parser_test.exs
defmodule Foo.ParserTest do
  use ExUnit.Case, async: true

  import Foo.Parser

  test "Number" do
    assert parse!("-10") == -10
    assert parse!("12") == 12
    assert parse!("0.1") == 0.1
    assert parse!("-0.1") == -0.1
  end

  test "List" do
    assert parse!("[]") == []
  end
end

File: books/the_elements_of_computing_systems/projects/tecs/src/test/scala/tecs/vm/ParserTest.scala
package tecs.vm

import org.scalatest.FunSpec

class ParserTest extends FunSpec {
  val p = Parser()

  describe("#parse") {
    it("commnd") {
      assert(p.parse("eq") == Eq())
    }

    it("label IF_TRUE") {
      // assert(p.parse("label IF_TRUE") == Command("label", Some("IF_TRUE"), None))
    }

    it("push constant 17") {
      assert(p.parse("push constant 17") == Push("constant", "17"))
    }
  }
}
